community/planning/batch_submitter.md

```diff
@@ -1,86 +1,121 @@
-the initial query for unclaimed batches are a part of the same database
-transaction.
-
-3. For each batch, the submitter will attempt to submit the batch. If the
-submitter receives an internal error or service unavailable error from the
-Splinter or Sawtooth network it will relinquish its claim to the batch, so
-that it or another submitter can retry sending the batch at a later interval.
-If The submitter receives a bad request error from the splinter network it will
-update the database by setting the submitted column to `true`, and updating the
-`submission_error` and `submission_error_message` columns describing why the
-batch was not submitted. If the batch was submitted successfully the submitted
-column is simply updated to `true`.
-
-4. The submitter will go to sleep for a configurable number of seconds and then
-repeat the process.
-
-## Database Schema
-
-```
-CREATE TABLE batches (
-    header_signature TEXT PRIMARY KEY,
-    data_change_id TEXT,
-    signer_public_key TEXT NOT NULL,
-    trace BOOLEAN NOT NULL,
-    serialized_batch TEXT NOT NULL,
-    submitted BOOLEAN NOT NULL,
-    submission_error VARCHAR(16),
-    submission_error_message TEXT,
-    dlt_status VARCHAR(16),
-    claim_expires TIMESTAMP,
-    created TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
-    service_id TEXT
-);
-
-CREATE TABLE transactions (
-    header_signature TEXT PRIMARY KEY,
-    batch_id TEXT NOT NULL,
-    family_name TEXT NOT NULL,
-    family_version TEXT NOT NULL,
-    signer_public_key TEXT NOT NULL,
-    FOREIGN KEY (batch_id) REFERENCES batches(header_signature) ON DELETE CASCADE
-);
-
-CREATE TABLE transaction_receipts (
-    id BIGSERIAL PRIMARY KEY,
-    transaction_id TEXT UNIQUE,
-    result_valid BOOLEAN NOT NULL,
-    error_message TEXT,
-    error_data TEXT,
-    serialized_receipt TEXT NOT NULL,
-    external_status VARCHAR(16),
-    external_error_message TEXT
-);
-```
+## Overview
+
+The batch submitter component manages the submission of batches to the
+underlying distributed ledger technology (DLT). It is responsible for providing
+the submission queuer and DLT monitor with the information they need to
+accurately perform their own functions via updates to the batch database.
+
+### Why must the submitter exist?
+
+* We need to submit batches via individual http requests to the DLT
+
+### How will the submitter be used?
+
+* The submitter will poll the batch queue for batches to submit
+* The submitter will update the batch database via the store
+
+### What must the submitter do?
+
+* Submit batches in the order in which it receives them from the queuer
+* Update a batch's status via the store as soon as possible
+* Handle and report connection issues with the DLT or service
+
+### What must the submitters not do?
+
+* Submit batches to the wrong service
+* Get hung up on a slow request
+
+### Design priorities
+
+1. Accuracy
+2. Stability
+3. Scalability
+4. HA
```

Comment:
is this a well-known term that we don't need to write it out completely ? I wouldn't really know what HA is if I just saw this

Reply:
Better to be more clear, thanks

Reply:
Now I can see it!